J arredondo enemy

JosephArr12 opened the pull request:
Enemies & Spawning System 

Three enemy types each with own script & animation controller. They will follow the player by setting their destination to player (via tag) and can deal damage. They can take damage themselves and can be killed.

<img width="733" alt="Screen Shot 2022-04-19 at 9 14 04 PM" src="https://user-images.githubusercontent.com/80712293/164150939-3086d6a3-c1b4-4ec3-953c-a1c87edd62a9.png">
<img width="723" alt="Screen Shot 2022-04-19 at 9 15 26 PM" src="https://user-images.githubusercontent.com/80712293/164150963-956d2f4c-3b5a-4617-bf4d-6677ee6b55b1.png">
<img width="721" alt="Screen Shot 2022-04-19 at 9 15 32 PM" src="https://user-images.githubusercontent.com/80712293/164151008-6aeb06b0-72cd-491e-9762-c51a97415ef6.png">

Enemies spawn at one of the points and player can advance to next wave by killing the remaining enemies.

JosephArr12 committed:
First commit

implemented three enemy types, spawning system and some UI elements such as wave number and hit effect.

JosephArr12 committed:
2nd commit

fixed some issues

JosephArr12 committed:
third commit

JosephArr12 committed:
First milestone commit

MasonJason23 commented:
Hey @JosephArr12 
This looks pretty nice and everything works fine on my end when I move to your branch, other than the merge conflicts 👍

JosephArr12 committed:
Added some effects(headshot), powerups

Powerup not implemented, enemies just drop one randomly when they die, and is destroyed once player "touches it" but nothing happens afterwards.

danielreyes2873 committed:
Removed TextMeshPro files

danielreyes2873 committed:
Removed TextMesh Pro and .json files

danielreyes2873 committed:
Merge branch 'main' into JArredondo_Enemy

danielreyes2873 commented:
Everything looks good to me. Sorry this PR took so long. I removed some conflicted files from certain commits with GitHub Desktop and manually resolved conflicts in other files. I kept a back up in case there are any problems. Merging now.
